packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -155,6 +143,28 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     onChangeProp.current = onChange;
   }, [onChange]);
 
+  // Record to store all extra information (e.g. original indexes), where the key is the id of the option
+  const optionsExtraInfoRef = React.useRef<
+    Record<string, { __originalIndex: number }>
+  >({});
+
+  // Clear the extra info when the options change so that it can be reinitialized below
+  React.useEffect(() => {
+    optionsExtraInfoRef.current = {};
+  }, [id, options]);
+
+  // Initialize the extra info only if it is not already initialized
+  if (
```

Comment:
Why this approach instead of `useEffect` ?

Reply:
if using `useEffect`, this ref is not available on first render so I can't access `__originalIndex` inside `getMenuItem`

Reply:
ref is available in useEffect, unless element is conditional.

Reply:
a wait, ref of menu items?

Reply:
`optionsExtraInfoRef` doesn't get set until after the useEffect, so it's not available when `getMenuItem` is first called